lib/domain/models/memo_collection_metadata.dart

```diff
@@ -0,0 +1,21 @@
+/// Fundamental fields that defines a `Memo` when relating to a `Collection`
+class MemoCollectionMetadata {
+  MemoCollectionMetadata({required this.uniqueId, required this.rawQuestion, required this.rawAnswer});
+
+  /// Identifies a global unique id (unique both in the parent's `Collection` and through all other `Memo`)
+  final String uniqueId;
```

Comment:
I confess that I'm a little bit confuse about this `unique` usage. By default, identifiers should be treated as unique, why do we need to specify that it's unique?

Reply:
Because we must reinforce this through the naming. One might think that a memo id could be unique only between its siblings in a collection, but no, it must be globally unique between all other memos

Reply:
IMO you're over engineering this definition. It's easier to assume that all identifiers are unique by default, instead reinforcing it in the property name and in its description. As an example, the collections identifiers are also unique, but their id property stands only by `id`. If we someday decide to wrap the collections in another structure that holds a set of collections, we would not refactor this `id` property to reinforce that it's unique. We could add a note on README or somewhere else about this definition that all ids are unique, but in code will create a unnecessary complexity IMO

Reply:
IMO you're over engineering this definition. It's easier to assume that all identifiers are unique by default, instead reinforcing it in the property name and in its description. As an example, the collections identifiers are also unique, but their id property stands only by id. If we someday decide to wrap the collections in another structure that holds a set of collections, we would not refactor this id property to reinforce that it's unique. We could add a note on README or somewhere else about this definition that all ids are unique, but in code will create a unnecessary complexity IMO